### PR TITLE
Make retries configurable; align constants

### PR DESCRIPTION
**Pull request by mre**
Using the same default values for the library and the
binary now. Tweaked the values a bit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -233,11 +233,12 @@ OPTIONS:
     -f, --format <format>                      Output format of final status report (compact, detailed, json, markdown)
                                                [default: compact]
         --github-token <github-token>          GitHub API token to use when checking github.com links, to avoid rate
-                                               limiting [env: GITHUB_TOKEN=]
+                                               limiting [env: GITHUB_TOKEN=39cc0d59a69f0ac929174b52b3517216704fd977]
```
Comment (lebensterben) [inline README.md:236]: not a good idea to expose the token

Comment (mre) [reply README.md:236]: Ouch. I revoked the token and made sure that it wasn't used in the meantime according to the Github logs. Thanks for letting me know.

Comment (mre) [reply README.md]: Also added 
```
#[structopt(long, env = "GITHUB_TOKEN", hide_env_values = true)]
```
to make sure this won't happen again.

**Comment by mre** [comment]
Hum, the tests failed because the `test_quirks` is getting flaky. Specifially the Twitter one:

```
> lychee fixtures/TEST_QUIRKS.txt
Issues found in 1 input(s). Find details below.

[/path/to/fixtures/TEST_QUIRKS.txt]:
✗ https://twitter.com/zarfeblong/status/1339742840142872577: Network error while trying to connect to an endpoint via reqwest

🔍 3 Total ✅ 2 OK 🚫 1 Error (HTTP:1)
```

**Pull request merged**